src/main/java/org/sonarsource/sonarlint/ls/AnalysisManager.java

```diff
@@ -285,17 +283,27 @@ private void analyze(URI fileUri, boolean shouldFetchServerIssues) {
     return javaConfigOpt;
   }
 
-  private static IssueListener createIssueListener(Map<URI, PublishDiagnosticsParams> files) {
+  private IssueListener createIssueListener() {
     return issue -> {
       ClientInputFile inputFile = issue.getInputFile();
       if (inputFile != null) {
         URI uri = inputFile.getClientObject();
-        PublishDiagnosticsParams publish = files.computeIfAbsent(uri, AnalysisManager::newPublishDiagnostics);
-        convert(issue).ifPresent(publish.getDiagnostics()::add);
+        issuesPerFileURI.computeIfAbsent(uri, u -> new ArrayList<>()).add(issue);
       }
     };
   }
 
+  Optional<Issue> getIssueForDiagnostic(URI fileUri, Diagnostic d) {
```

Comment:
It would be safer to make use of a unique id here, when the data field will be supported on diagnostics.

Reply:
Yup, I created a [ticket](https://jira.sonarsource.com/browse/SLLS-24) to tackle in a future version.